compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -442,21 +447,25 @@ static BOOL_TYS: TyS<'static> = TyS {
     kind: ty::Bool,
     flags: TypeFlags::empty(),
     outer_exclusive_binder: DebruijnIndex::from_usize(0),
+    stable_hash: Fingerprint::ZERO,
 };
 
 impl<'a, 'tcx> HashStable<StableHashingContext<'a>> for Ty<'tcx> {
     fn hash_stable(&self, hcx: &mut StableHashingContext<'a>, hasher: &mut StableHasher) {
         let TyS {
-            ref kind,
+            kind,
 
             // The other fields just provide fast access to information that is
             // also contained in `kind`, so no need to hash them.
             flags: _,
 
             outer_exclusive_binder: _,
+
+            stable_hash,
         } = self.0.0;
 
-        kind.hash_stable(hcx, hasher);
+        assert_ne!(*stable_hash, Fingerprint::ZERO, "{:#?}", kind);
```

Comment:
I've seen `assert_eq!` and `assert_ne!` produce a lot more machine code than a simple `assert!`. Not sure if that can influence performance in this case.

Reply:
Oh yea, this is all very hacky and naive. I just wanted to see if it's remotely interesting to do